clang/lib/CodeGen/CGOpenMPRuntimeGPU.cpp

```diff
@@ -752,27 +751,41 @@ void CGOpenMPRuntimeGPU::emitNonSPMDKernel(const OMPExecutableDirective &D,
   EntryFunctionState EST;
   WrapperFunctionsMap.clear();
 
+  bool IsBareKernel = D.getSingleClause<OMPXBareClause>();
```

Comment:
How can a non-spmd kernel be bare? We should force bare kernels to be SPMD, make this an assertion.

Reply:
So `target teams` is generic mode. Since we don't emit the kernel init/deinit, the kernel env will not be emitted as well. At runtime we will default to SPMD mode.
As discussed offline, we will need an extra mode, if we want, but no kernel env should work perfectly fine.

Reply:
I made the change in the parent scope such that a bare kernel is taken as SPMD as well.